### PR TITLE
8.0 port module product_serial

### PR DESCRIPTION
**Pull request by alexis-via**
Module product_serial is now available for v8 ! At last !

The code has been simplified a lot, because with the new stock management in v8, it is much simpler.

For the "Spread serial numbers" wizard, I took the latest code from my (unmerged) merge proposal on LP:
https://code.launchpad.net/~akretion-team/stock-logistic-flows/70-product_serial-plus-plus

### CODE REVIEW

File: product_serial/README.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+On the transfer wizard, for the products with *Lot Split Type* = *Single*, you will automatically get one line per unit (i.e. if the stock move has qty = 3, you will get 3 lines with qty = 1).
+
+This module also adds a button *Spread Serial Numbers* on the transfer wizard which make it easy to create a sequence of serial numbers or import serial numbers from a file.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/154/8.0
+
+For further information, please visit:
```
Comment (elicoidal) [inline product_serial/README.rst]: Hi @alexis-via Thanks for your contribution
You can remove the forum section

File: product_serial/README.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+------------
+
+* Alexis de Lattre <alexis.delattre@akretion.com>
+
+Maintainer
+----------
+
+.. image:: http://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: http://odoo-community.org
```
Comment (elicoidal) [inline product_serial/README.rst]: https (x2)

File: product_serial/__init__.py [hunk 3]
```diff
@@ -1,25 +1,4 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline product_serial/__init__.py:1]: Please use simple headers in all py files following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: product_serial/wizard/prodlot_wizard.py [hunk 4]
```diff
@@ -22,116 +22,150 @@
...
+
+    product_id = fields.Many2one(
+        'product.product', string='Product', required=True)
+    prefix = fields.Char(string='Prefix')
+    suffix = fields.Char(string='Suffix')
+    first_number = fields.Char(string='First Number')
+    last_number = fields.Char(string='Last Number')
+    prodlot_file = fields.Binary(
+        string='Serial Numbers File',
+        help="The serial numbers file should be a text file with one line "
```
Comment (elicoidal) [inline product_serial/wizard/prodlot_wizard.py]: s/text/csv(?)
You should detail the file structure (here or in README) or provide an example in the module.

File: product_serial/wizard/prodlot_wizard.py [hunk 5]
```diff
@@ -22,116 +22,150 @@
...
+                        % (current_lot,
+                           transfer_item.product_id.name_get()[0][1]))
+                lot = lots[0]
+
+            transfer_item.lot_id = lot.id
+        return transfer.wizard_view()
+
+    @api.multi
+    def cancel(self):
+        """We have to re-call the wizard when the user clicks on Cancel"""
```
Comment (elicoidal) [inline product_serial/wizard/prodlot_wizard.py]: The wizard is re-called  ...

**Comment by alexis-via** [comment]
@elicoidal I took into account all your remarks

**Comment by elicoidal** [comment]
LGTM (no technical review)

**Comment by AlexandreFournier** [comment]
+1

**Comment by AlexandreFournier** [comment]
+1

**Comment by Cedric-Pigeon** [comment]
@alexis-via Hello, do you plan to achieve this PR ? Is it still usefull ? Otherwise it will be closed in 3 days. Thanks.

**Comment by rousseldenis** [comment]
@alexis-via

**Comment by alexis-via** [comment]
For me, it's ready to be merged. All remarks have been taken into account.
I made a rebase.

**Comment by rousseldenis** [comment]
@alexis-via ok, awaiting travis

**Review by rousseldenis — APPROVED** [review]
Code review.

**Comment by rousseldenis** [comment]
As @elicoidal remarks have been attended, merging

**Pull request merged**